packages/ckeditor5-paste-from-office/src/filters/removemsattributes.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/removestyleblock
+ */
+
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Cleanup MS garbage like styles, attributes and elements.
+ *
+ * @param documentFragment element `data.content` obtained from clipboard.
+ */
+export default function removeMSAttributes( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	const elementsToRemove = [];
+
+	for ( const { item } of writer.createRangeIn( documentFragment ) ) {
+		if ( !item.is( 'element' ) ) {
+			continue;
+		}
+
+		const itemClassName = item.getAttribute( 'class' );
+
+		if ( itemClassName && /\bmso/gi.exec( itemClassName ) ) {
+			writer.removeAttribute( 'class', item );
+		}
+
+		if ( Array.from( item.getStyleNames() ).length ) {
+			for ( const styleName of item.getStyleNames() ) {
+				if ( /\bmso/gi.exec( styleName ) ) {
+					writer.removeStyle( styleName, item );
+				}
+			}
+		}
+
+		if ( item.is( 'element' ) && item.name === 'w:sdt' ) {
+			if ( item.is( 'element' ) ) {
+				elementsToRemove.push( item );
+			}
+		}
+	}
+
+	for ( const item of elementsToRemove ) {
+		const firstChild = item.getChild( 0 );
+
+		if ( firstChild && firstChild.is( 'element' ) ) {
+			writer.replace( item, firstChild );
+		}
```

Comment:
This could lead to content loss. Why are you handling only the first child and only when there is an element?